### ISSUE TITLE
Support instrumental variables in Probit/Logit regressions

### ISSUE THREAD

**Issue by groverlynn** [lrberge/fixest#320]
Can you please support the IV regression for probit and logit in `feglm`? The basic idea for getting the IV estimators is to include the residuals from the first stage OLS into the second stage probit/logit (which is the same for linear 2SLS to get the correct standard errors in one step). Probit/Logit needs one more step to correct the standard errors. There's the [Newey (1987)](https://doi.org/10.1016/0304-4076(87)90001-7) correction for probit (implemented in Stata `ivprobit`) and [Terza (2016)](https://doi.org/10.1177%2F1536867X1601600206) correction for both probit and logit. 

[Palmer et al. (2017)](https://doi.org/10.1093/aje/kwx175) has the simple R codes for probit and logit using the `glm` function.

**Comment by etiennebacher** [comment]
Hello, I don't think this will be addressed, see https://github.com/lrberge/fixest/issues/302

**Comment by lrberge** [comment]
Hi, thanks for the link to the paper which is really clear.
From skimming it: does it mean that the regular sandwich corrections can be applied with the Newey correction? If so: can this framework be extended to other families (e.g. Poisson)?

If true to the above, I may include it indeed. But it's quite long run since a) my todo list regarding this package is already quite long and b) I've completely stopped working on it during my workdays (to finally do my real job!) so my dev time is very limited indeed.

**Comment by groverlynn** [comment]
> From skimming it: does it mean that the regular sandwich corrections can be applied with the Newey correction?

Yes. The Newey correction incorporates both the variability explained in the estimate by the instrumental
variable and the predicted first stage residuals (unmeasured confounders). So, all sandwich vcov corrections can be used, as long as all regression in both stages use the same sandwich vcov correction.

> If so: can this framework be extended to other families (e.g. Poisson)?

According to Palmer et al. (2017), both the Newey correction and the Terza correction can be applied to Gaussian (linear), Probit, Logit, Poisson, and Gamma. It should work with all glm families.

**Comment by lrberge** [comment]
Thanks! My main issue with this kind of IV was the absence (due to my limited knowledge [*btw I don't use these models*]) of analytical standard errors.

But now that this problem seems solved I see so no more reason why it should not be included in the package.

And I *really* **really** like the generality (that it applies to all GLM). I must say I'm even a bit excited about it! 

I'll try to push for V1.0.0 by the end of the summer and hope I can include it.

**Comment by simonschoe** [comment]
@lrberge on a related tangent: Let's say I'd like to run a first-stage Probit and a second stage (classic) OLS, is this currently doable with how `feols` is set up?

**Comment by adamaltmejd** [comment]
Just ran into a situation where I need to use a logit IV. Do you have any progress update on this @lrberge? The alternatives in R do not seem plentiful.

**Comment by lrberge** [comment]
@simonschoe: no this is not in place, you would need to do it manually

@adamaltmejd: I thought it was simple but in fact it's not (as usual). From what I remember to compute the SEs of the coefficients of interest you would need the SEs of the fixed-effects (and a product with that potentially very large matrix), which is a no go. One may apply simplifications and may get around that problem but it needs to be worked out.

**Comment by adamaltmejd** [comment]
Thanks for the update, thats unfortunate!
